sys/net/gnrc/netif/Kconfig

```
@@ -22,12 +22,12 @@ config GNRC_NETIF_MSG_QUEUE_SIZE_EXP
 
 config GNRC_NETIF_IPV6_ADDRS_NUMOF
     int "Maximum number of unicast and anycast addresses per interface"
-    default 2
+    default 3
```

@leandrolanzieri, is something like

```suggestion
    default 2 + CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX if CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX
    default 2
```

possible?

---

I'm afraid not, no arithmetic is possible there. Probably some logic with macros should be added on the header file based on these options.

---

But 

```suggestion
    default 3 if CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX
    default 2
```

could be possible as a fallback, right?

---

I think `default 3 if DHCPV6_CLIENT_ADDR_LEASE_MAX = 0`.
BTW the `CONFIG_` prefix should not be used in `Kconfig` files, it is added later on for macros and makefiles.

Also, one has to consider this caveat https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html?highlight=kconfig#stuck-symbols-in-menuconfig-and-guiconfig. Perhaps a switch is wanted to hide the prompt, letting the defaults work, and if the user wants a specific value, it enables it.

```

config GNRC_NETIF_IPV6_ADDRS_NUMOF
    int
    prompt "Maximum number of unicast and anycast addresses per interface" if GNRC_NETIF_IPV6_ADDRS_CUSTOM_NUMOF
    default 3 if DHCPV6_CLIENT_ADDR_LEASE_MAX = 0
    default 2
```

---

> I think `default 3 if DHCPV6_CLIENT_ADDR_LEASE_MAX = 0`.

You mean `= 1`, right? Otherwise, why increase the number of unicast addresses, if there are no DHCPv6 leases for more addresses.

---

> > I think `default 3 if DHCPV6_CLIENT_ADDR_LEASE_MAX = 0`.
> 
> You mean `= 1`, right? 

Yes, sorry about that.

---

Or actually `!= 0`

---

I added your suggestions now in e2ff07b6de2310b3f8f366c281da2ee7dacb3936. I omitted the check for `GNRC_NETIF_IPV6_ADDRS_CUSTOM_NUMOF` for now, however – should such an option also be added? I also reverted the change of the help text regarding the default values – I assume there is no way to also determine the contents of the help text conditionally?

---

Can you have a look please, @leandrolanzieri?

---

It looks OK, and the default value will be applied when, for example, the `app.config` contains `CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX=0`. Still, the effect of "stuck" symbol in menuconfig will occur.